fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -179,6 +183,47 @@ private IndexingByIndex getIndexerByIndex() {
         if (ex == null) {
             return AsyncUtil.DONE;
         }
+        if (ex.getCause() instanceof IndexingBase.PartlyBuiltException) {
```

Comment:
It seems a little brittle to assume that exception's cause (but not the exception but not the exception's cause's cause, etc) is the `PartlyBuiltException`, especially as various things can wrap or unwrap an exception. I think I'd be more comfortable with something (probably a helper method) that walked up the cause chain and found the appropriate exception, like `FDBExceptions::getFDBCause` does:

https://github.com/Foundationdb/fdb-record-layer/blob/77efbbb8f3f65e7113fb31af1ae41e1da60c01b7/fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBExceptions.java#L228

Reply:
Fixed. 
`partlyBuiltException = IndexingBase.getAPartlyBuildExceptionIfApplicable(ex)`